code/atom.dm

```diff
@@ -584,6 +584,11 @@
 	if (iscarbon(usr) || issilicon(usr))
 		add_fingerprint(usr)
 
+	if (istype(usr, /mob/living/critter/flock/drone))
+		var/mob/living/critter/flock/drone/flockdrone = usr
+		if (flockdrone.floorrunning)
+			return
```

Comment:
This feels gross but I'm fine with it if doing it another way is a pain. Assuming they aren't supposed to be able to pick things up, etc while floor running, can you just mark them as restrained or something to prevent clicks in general?

Reply:
Yeah, they aren't supposed to be able to. We implemented a click() for Flockdrones to prevent doing stuff while floorrunning, but you could get around the ctrl+click pulling by right clicking and selecting pull. Sounds like restrained would be better?